apps/crn-server/src/controllers/research-outputs.ts

```diff
@@ -212,6 +226,23 @@ export default class ResearchOutputs implements ResearchOutputController {
       },
     });
   }
+
+  private async associateResearchOutputToAuthors({
+    id,
+    name,
+  }: ExternalAuthorPostRequest) {
+    if (id) return id;
+
+    if (name) {
+      const { id: authorId } =
+        await this.externalAuthorSquidexRestClient.create(
+          parseToSquidex({ name }) as RestExternalAuthor['data'],
```

Comment:
any reason why you have to cast it?

Reply:
in fact - i know why - its because of parseToSquidex which is poorly typed. I would make sure to type it in such a way that the output depends on the input, that way we can keep the type safety

Reply:
Fair point, thanks. Actually, we don't need `parseToSquidex` for this. It's only handling one prop `name`
https://github.com/yldio/asap-hub/blob/09c69b028726d8bbd6d6dcb63e1250fca0b7a7b1/apps/crn-server/src/controllers/research-outputs.ts#L238-L240

Reply:
ok basically what i mean is that the input type should enable posting:
1) new external authors 
2) references to external authors
3) references to authors

What I just realised is that actually we don't have to specify which of the two types of resources (ie external author or user) we are giving to squidex - and I now wonder if it should stay that way? I was going to suggest that since we change the interface we should change it to something like:

```typescript
{
(...)
authors?: Array<
    | { externalAuthorId: string }
    | { externalAuthorName: string }
    | { userId: string }
  >;
}
```
Technically we could instead just do:
```typescript
{
(...)
authors?: Array<
    | { id: string }
    | { userId: string }
  >;
}
```
And that's pretty much what you suggested (but not quite!) but I'm tempted to make the clients give us the type of resource they are associating here. What do you think? 

Also CC @markcsims @kmaid

Reply:
So an object like?

`{ id?: string, name: string, type: string}[]`

Reply:
@peterstarling @markcsims ok, so we'll change the type of the authors' field for something like this
```
{
     id?: string       // Optional because new external authors don't have an id
     name: string
     type: NewExternalAuthor | ExternalAuthor | InternalUser
}
```

Reply:
I'm still for this:

(...)
authors?: Array<
    | { externalAuthorId: string }
    | { externalAuthorName: string }
    | { userId: string }
  >;